bindings/java/src/main/rust/lib.rs

```diff
@@ -0,0 +1,126 @@
+use core::fmt;
+use css_inline::{CSSInliner, StylesheetCache};
+use jni::{
+    JNIEnv,
+    errors::Result as JNIResult,
+    objects::{JClass, JObject, JString},
+    sys::jstring,
+};
+use std::{borrow::Cow, num::NonZeroUsize};
+
+trait JNIExt {
+    fn get_bool_field(&mut self, obj: &JObject, name: &str) -> JNIResult<bool>;
+    fn get_int_field(&mut self, obj: &JObject, name: &str) -> JNIResult<i32>;
+    fn get_string_field_opt(&mut self, obj: &JObject, name: &str) -> JNIResult<Option<String>>;
+}
+
+impl<'a> JNIExt for JNIEnv<'a> {
+    fn get_bool_field(&mut self, obj: &JObject, name: &str) -> JNIResult<bool> {
+        self.get_field(obj, name, "Z")?.z()
+    }
+
+    fn get_int_field(&mut self, obj: &JObject, name: &str) -> JNIResult<i32> {
+        self.get_field(obj, name, "I")?.i()
+    }
+
+    fn get_string_field_opt(&mut self, cfg: &JObject, name: &str) -> JNIResult<Option<String>> {
+        let value = self.get_field(cfg, name, "Ljava/lang/String;")?.l()?;
+        if value.is_null() {
+            Ok(None)
+        } else {
+            Ok(Some(self.get_string(&JString::from(value))?.into()))
+        }
+    }
+}
+
+enum Error<E> {
+    Jni(jni::errors::Error),
+    Other(E),
+}
+
+impl<E> From<jni::errors::Error> for Error<E> {
+    fn from(value: jni::errors::Error) -> Self {
+        Error::Jni(value)
+    }
+}
+
+impl<E: fmt::Display> fmt::Display for Error<E> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Error::Jni(error) => error.fmt(f),
+            Error::Other(error) => error.fmt(f),
+        }
+    }
+}
+
+fn build_inliner(
+    env: &mut JNIEnv,
+    cfg: JObject,
+) -> Result<CSSInliner<'static>, Error<css_inline::ParseError>> {
+    let inline_style_tags = env.get_bool_field(&cfg, "inlineStyleTags")?;
+    let keep_style_tags = env.get_bool_field(&cfg, "keepStyleTags")?;
+    let keep_link_tags = env.get_bool_field(&cfg, "keepLinkTags")?;
+    let load_remote_stylesheets = env.get_bool_field(&cfg, "loadRemoteStylesheets")?;
+    let cache_size = env.get_int_field(&cfg, "cacheSize")?;
+    let preallocate_node_capacity = env.get_int_field(&cfg, "preallocateNodeCapacity")?;
+
+    let extra_css = env.get_string_field_opt(&cfg, "extraCss")?;
+    let base_url = env.get_string_field_opt(&cfg, "baseUrl")?;
+    let mut builder = CSSInliner::options()
+        .inline_style_tags(inline_style_tags)
+        .keep_style_tags(keep_style_tags)
+        .keep_link_tags(keep_link_tags)
+        .load_remote_stylesheets(load_remote_stylesheets)
+        .extra_css(extra_css.map(Cow::Owned))
+        .preallocate_node_capacity(preallocate_node_capacity as usize);
```

Comment:
Use `usize::try_from` + return an error (even if there validation on the Java side)